html/pfappserver/root/static.alt/src/globals/pfConfigurationSwitches.js

```diff
@@ -0,0 +1,2010 @@
+  port: {
+    value: 'port',
+    text: i18n.t('Port'),
+    types: [fieldType.INTEGER],
+    validators: {
+      type: {
+        /* Don't mix with 'always' */
+        [i18n.t('Condition conflicts with "Always".')]: restrictAllSiblingFields('type', 'always')
+      },
+      value: {
+        [i18n.t('Value required.')]: required,
+        [i18n.t('Invalid Port Number.')]: isPort,
+        /* Don't allow duplicates */
+        [i18n.t('Duplicate Port.')]: limitSiblingFields(['type', 'value'])
+      }
+    }
+  },
+  mac: {
+    value: 'mac',
+    text: i18n.t('MAC Address'),
+    types: [fieldType.SUBSTRING],
+    validators: {
+      type: {
+        /* Don't mix with 'always' */
+        [i18n.t('Condition conflicts with "Always".')]: restrictAllSiblingFields('type', 'always')
+      },
+      value: {
+        [i18n.t('Value required.')]: required,
+        [i18n.t('Invalid MAC Address.')]: macAddress,
+        /* Don't allow duplicates */
+        [i18n.t('Duplicate MAC Address.')]: limitSiblingFields(['type', 'value'])
+      }
+    }
+  },
+  ssid: {
+    value: 'ssid',
+    text: i18n.t('Wi-Fi Network SSID'),
+    types: [fieldType.SUBSTRING],
+    validators: {
+      type: {
+        /* Don't mix with 'always' */
+        [i18n.t('Condition conflicts with "Always".')]: restrictAllSiblingFields('type', 'always')
+      },
+      value: {
+        [i18n.t('Value required.')]: required,
+        /* Don't allow duplicates */
+        [i18n.t('Duplicate SSID.')]: limitSiblingFields(['type', 'value'])
+      }
+    }
+  }
+}
+
+export const pfConfigurationSwitchViewFields = (context = {}) => {
+  let {
+    isNew = false,
+    isClone = false,
+    placeholders = {}, // form placeholders
+    switche = {}, // form
+    roles = [], // all roles
+    switchGroups = [] // all switchGroups
+  } = context
+
+  return [
+    {
+      tab: i18n.t('Definition'),
+      fields: [
+        {
+          label: i18n.t('IP Address/MAC Address/Range (CIDR)'),
+          fields: [
+            {
+              key: 'id',
+              component: pfFormInput,
+              attrs: {
+                disabled: (!isNew && !isClone)
+              },
+              validators: {
+                [i18n.t('Identifier required.')]: required,
+                [i18n.t('IP addresses only.')]: ipAddress,
+                [i18n.t('Maximum 255 characters.')]: maxLength(255)
+              }
+            }
+          ]
+        },
+        {
+          label: i18n.t('Description'),
+          fields: [
+            {
+              key: 'notes',
+              component: pfFormInput,
+              attrs: {
+                placeholder: ('notes' in placeholders) ? placeholders.notes : null
+              },
+              validators: {
+                [i18n.t('Description required.')]: required,
+                [i18n.t('Maximum 255 characters.')]: maxLength(255)
+              }
+            }
+          ]
+        },
+        {
+          label: i18n.t('Type'),
```

Comment:
We eventually need an endpoint to fetch all switch types. Do we have an issue opened regarding this?

Reply:
https://github.com/inverse-inc/packetfence/issues/3845